src/frontend/expr.rs

```diff
@@ -36,80 +73,264 @@ impl Expr {
+            Expr::Or(terms) => terms.iter().any(|c| c.is_aggregating(aggregating_funcs)),
+            Expr::Bool(_) => false,
+            Expr::BinaryOp { left, right, op: _ } => {
+                left.is_aggregating(aggregating_funcs) | right.is_aggregating(aggregating_funcs)
+            }
+            Expr::HasLabel(_, _) => false,
         }
     }
 }
 
-#[derive(Debug, PartialEq)]
+#[derive(Debug, PartialEq, Clone)]
 pub struct MapEntryExpr {
     pub key: Token,
     pub val: Expr,
 }
 
 pub(super) fn plan_expr(pc: &mut PlanningContext, expression: Pair<Rule>) -> Result<Expr> {
+    let mut or_expressions = Vec::new();
     for inner in expression.into_inner() {
         match inner.as_rule() {
-            Rule::string => {
-                let content = inner
-                    .into_inner()
-                    .next()
-                    .expect("Strings should always have an inner value")
-                    .as_str();
-                return Ok(Expr::Lit(Val::String(String::from(content))));
-            }
-            Rule::id => {
-                let tok = pc.tokenize(inner.as_str());
-                return Ok(Expr::Slot(pc.get_or_alloc_slot(tok)));
-            }
-            Rule::prop_lookup => {
-                let mut prop_lookup = inner.into_inner();
-                let prop_lookup_expr = prop_lookup.next().unwrap();
-                let base = match prop_lookup_expr.as_rule() {
-                    Rule::id => {
-                        let tok = pc.tokenize(prop_lookup_expr.as_str());
-                        Expr::Slot(pc.get_or_alloc_slot(tok))
-                    }
-                    _ => unreachable!(),
-                };
-                let mut props = Vec::new();
-                for p_inner in prop_lookup {
-                    if let Rule::id = p_inner.as_rule() {
-                        props.push(pc.tokenize(p_inner.as_str()));
-                    }
+            Rule::and_expr => {
+                let mut and_expressions: Vec<Expr> = Vec::new();
+                for term in inner.into_inner() {
+                    and_expressions.push(plan_term(pc, term)?)
                 }
-                return Ok(Expr::Prop(Box::new(base), props));
-            }
-            Rule::func_call => {
-                let mut func_call = inner.into_inner();
-                let func_name_item = func_call
-                    .next()
-                    .expect("All func_calls must start with an identifier");
-                let name = pc.tokenize(func_name_item.as_str());
-                // Parse args
-                let mut args = Vec::new();
-                for arg in func_call {
-                    args.push(plan_expr(pc, arg)?);
+                if and_expressions.len() == 1 {
+                    or_expressions.push(and_expressions.remove(0))
+                } else {
+                    or_expressions.push(Expr::And(and_expressions))
                 }
```

Comment:
Could use if as an expression

```suggestion
                let and_expr = if and_expressions.len() == 1 {
                    and_expressions.remove(0)
                } else {
                    Expr::And(and_expressions)
                };
                or_expressions.push(and_expr);
```